### PR TITLE
fix(client): set fcc-sound value in local storage after user data is fetched

### LINKED ISSUE #46372: Campfire Mode doesn't trigger on multiple browsers unless you turn it on and off
### Describe the Issue

Campfire Mode was not playing on my account. I did find a fix, but still wanted to report! Please note it could be affecting more than one page, but I only tested on the new responsive web design! 

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/

### Steps to Reproduce

1. I'm logged in on my computer 
2. Under settings, Campfire Mode is enabled. 
3. My volume is turned up!
4. I go to the (New) Responsive Web Design course
5. The serenade of the Guitar does not play as I type or click on items :(

### Expected behavior

1. I'm logged in on my computer 
2. Under settings, Campfire Mode is enabled. 
3. My volume is turned up!
4. I go to the (New) Responsive Web Design course
5. The serenade of the Guitar plays!

### Screenshots

_No response_

### System

- Mac OS Big Sur 11.6.5 running Safari
- Windows 10 running Google Chrome 

### Additional context

I found a fix while poking around! 

1. I go to my settings and turn off and on Campfire mode.
2. The serenade of the Guitar DOES play as I type and click on items :)

### PR DESCRIPTION
**Pull request by nayabatir1**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #46372
Closes #48107 

This is first method of solving issue
After user has logged in successfully I'm setting `fcc-sound` value in local storage after user data is fetched in `fetchSessionUser`.
As `playTone` grabs `fcc-sound` value from local storage to play audio

### CODE REVIEW

**Comment by ieahleen** [comment]
Please take a moment to give a name to the PR that follows the guidelines: https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title

**Comment by nayabatir1** [comment]
> Please take a moment to give a name to the PR that follows the guidelines: https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title

I've updated pr title.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Hi, could you limit the scope of this PR? If we can remove Tone, that's great, and we really appreciate you looking into this. However, each PR should ideally do one thing so that we (as maintainers) can easily evaluate each change.

Could you create a separate PR with just the removal of  Tone?

**Comment by raisedadead** [comment]
This PR need NOT be reviewed any further until we have achieved agreement on #48107

File: client/package.json [hunk 1]
```diff
@@ -122,7 +122,6 @@
     "sha-1": "1.0.0",
     "store": "2.0.12",
     "stream-browserify": "3.0.0",
-    "tone": "14.7.77",
```
Comment (naomi-lgbt) [inline client/package.json:125]: This needs to be reverted.

File: client/src/redux/fetch-user-saga.js [hunk 2]
```diff
@@ -19,6 +19,13 @@ function* fetchSessionUser() {
       data: { user = {}, result = '', sessionMeta = {} }
     } = yield call(getSessionUser);
     const appUser = user[result] || {};
+
+    const [userId] = Object.keys(user);
+
+    const sound = user[userId].sound;
+
+    localStorage.setItem('fcc-sound', sound);
```
Comment (naomi-lgbt) [inline client/src/redux/fetch-user-saga.js:27]: Should we be using `store` here instead?

File: client/src/utils/tone/index.ts [hunk 3]
```diff
@@ -62,20 +62,16 @@ type ToneStates = keyof typeof toneUrls;
...
   const playSound = !!store.get('fcc-sound');
   if (playSound && toneUrls[state]) {
-    const tone = await import('tone');
+    const Tone = await import('tone');
 
-    if (tone.context.state !== 'running') {
-      tone.context.resume().catch(err => {
-        console.error('Error resuming audio context', err);
-      });
-    }
```
Comment (naomi-lgbt) [inline client/src/utils/tone/index.ts:71]: Why is this removed? We need to check that the audiocontext state is running, and if not, resume it.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: client/src/utils/tone/index.ts (hunk 3 above)
Comment (nayabatir1) [reply client/src/utils/tone/index.ts:71]: reference [doc](https://tonejs.github.io/docs/14.7.77/Context#resume)
I'm using `player.start()` to play audio which is basically same thing. Start audio context from suspended state

File: client/src/redux/fetch-user-saga.js (hunk 2 above)
Comment (nayabatir1) [reply client/src/redux/fetch-user-saga.js:27]: I don't understand what additional benefits `store` provide over `localStorage` api, also syntactically both are similar.

Comment (naomi-lgbt) [reply client/src/redux/fetch-user-saga.js:27]: I believe `store` will adapt to use other storage APIs if the user is using a non-modern browser.

**Comment by Sembauke** [comment]
Hey @nayabatir1,

Are you still working on this? Do you need help?

**Comment by nayabatir1** [comment]
Hey @Sembauke,
I already did all the changes. If I missed something do let me know.
